src/python-common/ceph/deployment/service_spec.py

```diff
@@ -1095,3 +1113,106 @@ def get_port(self) -> int:
             return {'prometheus': 9095,
                     'node-exporter': 9100,
                     'grafana': 3000}[self.service_type]
+
+
+class SNMPGatewaySpec(ServiceSpec):
+
+    def __init__(self,
+                 service_type: str = 'snmp-gateway',
+                 snmp_version: Optional[str] = None,
+                 snmp_destination: Optional[str] = None,
+                 credentials: Optional[Dict[str, str]] = None,
```

Comment:
do we know which keys can be in the credentials?

Reply:
yes its in the validate function - but the credentials vary by snmp type - that's why I left it to the validate instead of using something like enum. the requirements for v2c is just community and for v3 credentials are username/password and optionally a privacy password